sql/core/src/main/java/org/apache/spark/sql/execution/UnsafeKVExternalSorter.java

```diff
@@ -122,11 +119,7 @@ public UnsafeKVExternalSorter(
         /* initialSize */ 4096,
         pageSizeBytes,
         inMemSorter);
-
-      sorter.spill();
-      map.free();
-      // counting the memory used UnsafeInMemorySorter
-      taskMemoryManager.acquireExecutionMemory(inMemSorter.getMemoryUsage(), sorter);
+      map.reset();
```

Comment:
I think we need comments at here. Basically, we need to explain why we reset. Also, once `sorter` is created, `inMemSorter` will not be needed, right?